### PR TITLE
Add text support mode for campaign selection dialogs

### PR DESCRIPTION
**Pull request by ihhub**
Video file playback now does not require ROI to be added. This was a corner case only available for The Succession Wars campaign selection.

relates to #5270

### CODE REVIEW

File: src/fheroes2/game/game_hotkeys.cpp [hunk 1]
```diff
@@ -412,8 +412,16 @@ void Game::HotKeysDefaults( void )
...
+    key_events[EVENT_NEW_CAMPAIGN_SELECTION_SUCCESSION_WARS] = KEY_o;
+    key_events[EVENT_NEW_CAMPAIGN_SELECTION_PRICE_OF_LOYALTY] = KEY_e;
+
+    key_events[EVENT_NEW_ROLAND_CAMPAIGN] = KEY_1;
+    key_events[EVENT_NEW_ARCHIBALD_CAMPAIGN] = KEY_2;
+
+    key_events[EVENT_NEW_PRICE_OF_LOYALTY_CAMPAIGN] = KEY_1;
+    key_events[EVENT_NEW_VOYAGE_HOME_CAMPAIGN] = KEY_2;
+    key_events[EVENT_NEW_WIZARDS_ISLE_CAMPAIGN] = KEY_3;
+    key_events[EVENT_NEW_DESCENDANTS_CAMPAIGN] = KEY_4;
```
Comment (oleg-derevenetz) [inline src/fheroes2/game/game_hotkeys.cpp:424]: It would be nice to mention this to fheroes2.key as well.

**Review by oleg-derevenetz — CHANGES_REQUESTED** [review]

File: src/fheroes2/game/game_hotkeys.cpp (hunk 1 above)
Comment (ihhub) [reply src/fheroes2/game/game_hotkeys.cpp:360]: It is done. We should remove this file as generate it like we do for the configuration file as many entries are missing. As of now I polished the code and added new entries into the hotkey file.

File: src/fheroes2/game/game_hotkeys.cpp [hunk 2]
```diff
@@ -430,14 +374,9 @@ bool Game::HotKeyHoldEvent( const int eventID )
...
-const char * Game::getHotKeyNameByEventId( const int eventID )
+std::string Game::getHotKeyNameByEventId( const int eventID )
 {
-    auto it = keyName.find( key_events[eventID] );
-    if ( it != keyName.end() ) {
-        return it->second;
-    }
-
-    return "???";
+    return StringUpper( KeySymGetName( key_events[eventID] ) );
```
Comment (ihhub) [inline src/fheroes2/game/game_hotkeys.cpp:379]: SDL has the same logic of converting key values into human-readable string so this code was modified.

File: src/fheroes2/game/game_hotkeys.cpp [hunk 3]
```diff
@@ -27,276 +27,212 @@
...
 
+#include <array>
+#include <cassert>
 #include <map>
 
+namespace
+{
+    std::array<KeySym, Game::EVENT_LAST> key_events{ KEY_NONE };
+
+    const char * getEventName( const int eventId )
```
Comment (ihhub) [inline src/fheroes2/game/game_hotkeys.cpp:40]: Some entries were missing in this function.

**Review by oleg-derevenetz — APPROVED** [review]

**Pull request merged**